task/k8s/task.go

```diff
@@ -302,3 +299,14 @@ func (t *Task) GetKeyPair(ctx context.Context) (*ssh.DeterministicSSHKeyPair, er
 func (t *Task) GetIdentifier(ctx context.Context) common.Identifier {
 	return t.Identifier
 }
+
+// withEnv runs the specified action with the environment variables set.
+func withEnv(env map[string]string, action func(context.Context) error) func(context.Context) error {
```

Comment:
This is only needed because we de-facto use env vars instead of passing function params. This needs to be fixed ASAP

Reply:
Resolving in favor of https://github.com/iterative/terraform-provider-iterative/issues/648.